clusterctl/cmd/create_cluster.go

```diff
@@ -155,30 +153,9 @@ func parseMachinesYaml(file string) ([]*clusterv1.Machine, error) {
 }
 
 func getProvider(provider string) (clusterdeployer.ProviderDeployer, error) {
-	switch provider {
-	case "google":
-		return google.NewMachineActuator(google.MachineActuatorParams{})
-	case "vsphere":
-		return &vsphereAdapter{vsphere.NewDeploymentClient()}, nil
-	case "azure":
-		//Work being done at https://github.com/platform9/azure-provider
-		return nil, errors.New("Azure not yet implemented")
-	default:
-		return nil, fmt.Errorf("Unrecognized provider %v", provider)
+	provisioner, err := clustercommon.GetClusterProvisioner(provider)
+	if err != nil {
+		return nil, err
 	}
-}
-
-// Adapt the vsphere methods calls since gcp/vsphere are not on the same page.
-// Long term, these providers should converge or the need for a provider will go away.
-// Whichever comes first.
-type vsphereAdapter struct {
-	*vsphere.DeploymentClient
-}
-
-func (a *vsphereAdapter) GetIP(cluster *clusterv1.Cluster, machine *clusterv1.Machine) (string, error) {
-	return a.DeploymentClient.GetIP(machine)
-}
-
-func (a *vsphereAdapter) GetKubeConfig(cluster *clusterv1.Cluster, master *clusterv1.Machine) (string, error) {
-	return a.DeploymentClient.GetKubeConfig(master)
+	return provisioner.(clusterdeployer.ProviderDeployer), nil
```

Comment:
An unconditional cast here could panic (since nothing verifies the type registered). This should do a conditional cast, or the get func should return the proper type.

Reply:
adding a check here to prevent panic